apps/omstillingsstoenad-ui/src/components/felles/Datovelger.tsx

```diff
@@ -79,14 +79,17 @@ const Datovelger = ({
                     validate: (date) => !date || isValid(date),
                 }}
                 render={({ field: { onChange, value } }) => {
-                    const { datepickerProps, inputProps } = useDatepicker({
+                    const { datepickerProps, inputProps, setSelected, selectedDay } = useDatepicker({
                         fromDate: minDate ? parseDateMaxMin(minDate) : undefined,
                         toDate: maxDate ? parseDateMaxMin(maxDate) : undefined,
                         locale: i18n.language as DatePickerProps['locale'],
                         inputFormat: 'dd.MM.yyyy',
                         defaultSelected: value ? new Date(value) : undefined,
                         onDateChange: (date) => onChange(formatDate(date)),
                     })
+
+                    if (value === undefined && selectedDay !== undefined) setSelected(undefined)
```

Comment:
Noe jeg kom til å tenke på når jeg så denne snutten, er at `undefined` som verdi til form-elementer kan være noe av det som skaper trøbbel -- for controlled inputs forventer den å få en verdi satt som prop (altså `value !== undefined`), mens for uncontrolled forventer den å få `undefined`, og liker ikke helt å bytte mellom de. 

Så vet jeg ikke om det er kilden til problemet her, og det krever uansett en del å endre verdiene over alt til å følge slike regler, men verdt å ha i bakhodet hvis man ser nærmere på dette en annen gang.

Reply:
Enig at det ikke er helt ryddig, men grunnen til at det er brukt `undefined` her er fordi det er kun Date og `undefined` som er støtte av Aksel sin DatePicker.